packages/go_router_builder/lib/src/type_helpers.dart

```diff
@@ -263,6 +264,49 @@ class _TypeHelperEnum extends _TypeHelperWithHelper {
   bool _matchesType(DartType type) => type.isEnum;
 }
 
+/// A type helper for extension types.
+class _TypeHelperExtensionType extends _TypeHelper {
+  const _TypeHelperExtensionType();
+
+  @override
+  String _decode(
+      ParameterElement parameterElement, Set<String> pathParameters) {
+    final DartType paramType = parameterElement.type;
+    if (paramType.isNullableType && parameterElement.hasDefaultValue) {
+      throw NullableDefaultValueError(parameterElement);
+    }
+
+    final String stateValue =
+        'state.${_stateValueAccess(parameterElement, pathParameters)}';
+    final String castType;
+    if (paramType.isNullableType || parameterElement.hasDefaultValue) {
+      castType = '$paramType${paramType.isNullableType ? '' : '?'}';
+    } else {
+      castType = '$paramType';
+    }
+
+    final DartType extensionTypeErasure = paramType.extensionTypeErasure;
+    if (extensionTypeErasure.isDartCoreString) {
+      return '$stateValue as $castType';
+    }
+
+    final String parseTypeName =
```

Comment:
Should we guard against no primitive type? since this is following by a hardcode tryParse, may as well only allow certain numeric types?

Reply:
I think you are right in pointing this out. I restricted the type and handled it correctly. [61901a3](https://github.com/flutter/packages/pull/9458/commits/61901a349e48ebe5155f3bbde768844bd7c16d5d)